### PR TITLE
feat: Switch from nosetest to pytest

### PR DESCRIPTION
**Pull request by bkyryliuk**
Changed test runner from nosetests to pytest
Tox locally seems to be working just fine and contributor instructions don't need to change.

Changes:
1. changed runner from nosetests to pytest
2. added pytest.ini for global pytest controls
3. updated run.sh and tox not to compute the coverage
4. updated test classes to start with Test_ and not too have __init__ method

Naming conventions:
* https://docs.pytest.org/en/reorganize-docs/new-docs/user/naming_conventions.html#:~:text=File%20names%20should%20start%20or,Test%E2%80%9D%2C%20as%20in%20TestExample%20.&text=To%20change%20how%20pytest%20recognises,Advanced%20%2D%20configuring%20test%20discovery).

*Testing*
* Made sure that test coverage stays the same
* Validated that tests match to the nosetest via pytest --collect-only

### CODE REVIEW

**Review by john-bodley — APPROVED** [review]

**Comment by ktmud** [comment]
1. Code coverage is missing
   

2. Might want to remove `nosetests` from pip dependencies and clean [this](https://github.com/apache/incubator-superset/blob/master/setup.cfg#L36-L41) up as well.

**Comment by bkyryliuk** [comment]
> 1. Code coverage is missing
>    
> 2. Might want to remove `nosetests` from pip dependencies and clean [this](https://github.com/apache/incubator-superset/blob/master/setup.cfg#L36-L41) up as well.

@ktmud good catch. Will do the 1. As for point 2 - I changed nosetests there to the tool:pytest

**Comment by ktmud** [comment]
Ah, sorry for missing that. Are those nosetests options in `setup.cfg` also valid for pytest as well?

**Comment by bkyryliuk** [comment]
> Ah, sorry for missing that. Are those nosetests options in `setup.cfg` also valid for pytest as well?

@ktmud  taking into account that we want to modify the local tests vs CI behavior that section is probably not needed.
Can you take another look at the PR? 
I've modified it to run test coverage only in the python_tests.sh

File: scripts/python_tests.sh
Comment (ktmud) [inline scripts/python_tests.sh:26]: Do we want coverage for this file as well? I don't think it's needed but it was covered in previous setup.

**Review by ktmud — APPROVED** [review]
Code LGTM! Do you know why did Codecov say overall coverage dropped by 17.79%?

File: scripts/python_tests.sh
Comment (bkyryliuk) [reply scripts/python_tests.sh:26]: my preference is not to count it, to me this is more a fixture setup rather than an actual test.
However do not have a preference here.

**Comment by bkyryliuk** [comment]
> Code LGTM! Do you know why did Codecov say overall coverage dropped by 17.79%?
old run, will try to see how can I trigger it again

File: scripts/python_tests.sh
Comment (bkyryliuk) [reply scripts/python_tests.sh]: https://codecov.io/gh/apache/incubator-superset/pull/10177/commits - looks like it was the reason for dropped coverage - will keep playing with it

**Comment by bkyryliuk** [comment]
@ktmud and @john-bodley could you please take another look?
I've missed some tests in the original implementation and had to do extra changes to the test class names.

**Comment by ktmud** [comment]
Codecov seems to be stuck at "[waiting for CI to complete](https://codecov.io/gh/apache/incubator-superset/commit/6a695f9f62b25b601afdea50403c4717b2b56f6e)", just like in #10142 . If you can get it fixed, that'd be great. If not, we can probably just merge this as is and see if it's a recurring issue in the next PR.

There were also some deprecation warnings in the logs that might worth fixing (could be in another PR):

```
tests/security_tests.py::TestRolePermission::test_set_perm_slice
  /home/runner/work/incubator-superset/incubator-superset/tests/security_tests.py:425: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(slice.schema_perm, table.schema_perm)
```

**Review by villebro — APPROVED** [review]

**Review by dpgaspar — APPROVED** [review]

**Comment by bkyryliuk** [comment]
> Codecov seems to be stuck at "[waiting for CI to complete](https://codecov.io/gh/apache/incubator-superset/commit/6a695f9f62b25b601afdea50403c4717b2b56f6e)", just like in #10142 . If you can get it fixed, that'd be great. If not, we can probably just merge this as is and see if it's a recurring issue in the next PR.
> 
> There were also some deprecation warnings in the logs that might worth fixing (could be in another PR):
> 
> ```
> tests/security_tests.py::TestRolePermission::test_set_perm_slice
>   /home/runner/work/incubator-superset/incubator-superset/tests/security_tests.py:425: DeprecationWarning: Please use assertEqual instead.
>     self.assertEquals(slice.schema_perm, table.schema_perm)
> ```

@ktmud this is interesting, report is generated on their side: https://codecov.io/gh/apache/incubator-superset/pull/10177
however was not send to update the PR, will amend the PR and see if it fixes it.

**Pull request merged**
